### PR TITLE
metashell 2.0.0 (new formula)

### PR DESCRIPTION
**Pull request by ortho**
Interactive template metaprogramming shell for c++

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Test is failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/20669/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_metashell/

Otherwise: looks good!

**Comment by DomT4** [comment]
Mountain Lion build failed for the semi-standard reason:

```
-- Performing Test LLVM_NO_OLD_LIBSTDCXX
-- Performing Test LLVM_NO_OLD_LIBSTDCXX - Failed
CMake Error at cmake/modules/HandleLLVMOptions.cmake:34 (message):
  Host Clang must be able to find libstdc++4.7 or newer!
Call Stack (most recent call first):
  CMakeLists.txt:315 (include)


-- Configuring incomplete, errors occurred!
See also "/tmp/metashell-yrnqQk/metashell-2.0.0/templight/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/metashell-yrnqQk/metashell-2.0.0/templight/build/CMakeFiles/CMakeError.log".
```

**Comment by mikemcquaid** [comment]
Needs a `needs :cxx11` line.

**Comment by ortho** [comment]
I have no idea why test is failed especially on Yosemite. Test is always success on my side
Need help to find out what happen. 

Is it possible to run ./metashell in a console to view a result?

Do I missied some dependencies?

```
otool -L /usr/local/bin/metashell
    /usr/lib/libncurses.5.4.dylib (compatibility version 5.4.0, current version 5.4.0)
    /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1213.0.0)
    /usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.5)
    /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.0.0)
```

File: Library/Formula/metashell.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  homepage "https://github.com/sabel83/metashell"
+  url "https://github.com/sabel83/metashell/archive/v2.0.0.tar.gz"
+  sha1 "4dec47b6ee32cdf179b2eb297c289b296d3fba8f"
+
+  depends_on "cmake" => :build
+
+  needs :cxx11
+
+  def install
+    # Build internal Clang
```
Comment (mikemcquaid) [inline Library/Formula/metashell.rb]: Probably also needs a `ENV.cxx11`

File: Library/Formula/metashell.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  test do
+    IO.popen("#{bin}/metashell", "w+") do |pipe|
+      pipe.puts("#include <boost/mpl/int.hpp>")
+      pipe.puts("#include <boost/mpl/plus.hpp>")
+      pipe.puts("using namespace boost::mpl;")
+      pipe.puts("plus<int_<6>, int_<7>>::type")
+      pipe.close_write
+      assert pipe.read.include?("integral_c")
```
Comment (mikemcquaid) [inline Library/Formula/metashell.rb]: If you use `assert_match` then the error might be a bit more clear.

**Comment by ortho** [comment]
There are errors during build 20799 in Jenkins.
```
 > git checkout -f d0bbcdc3d95baa90fdb1a8e36b0716484a76b7d0
FATAL: Could not checkout null with start point d0bbcdc3d95baa90fdb1a8e36b0716484a76b7d0
```

What am I doing wrong?

**Comment by mikemcquaid** [comment]
@ortho Nothing! That's an annoying Jenkins bug. I've just restarted the build.

**Comment by ortho** [comment]
Everything is working fine now. Do you accept this formula?
And if it so. How long will it take to merge it in the main branch?

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**